src/extractor/ConfigurationExtractor.ts

```diff
@@ -0,0 +1,113 @@
+import { Package } from '../processor';
+import { ExportableConfiguration, ExportableCaretValueRule, ExportableRule } from '../exportable';
+import { fshtypes } from 'fsh-sushi';
+
+export class ConfigurationExtractor {
+  static canonicalRegex = /^https?:\/\/(\w+)\.[\w.]+\/(.*)$/;
+
+  static process(resources: Package): ExportableConfiguration {
+    // canonical and fhirVersion are required elements in configuration
+    const canonical = ConfigurationExtractor.inferCanonical(resources);
+    const fhirVersion = ConfigurationExtractor.inferString(resources, 'fhirVersion');
+    const config = new ExportableConfiguration({
+      canonical: canonical,
+      fhirVersion: [fhirVersion]
+    });
+    // infer id and name using canonical
+    const canonicalMatch = canonical.match(ConfigurationExtractor.canonicalRegex);
+    if (canonicalMatch) {
+      const idParts = [canonicalMatch[1], ...canonicalMatch[2].split('/')];
+      config.config.id = idParts.join('.');
+      config.config.name = idParts.map(capitalize).join('');
+    }
+    // infer status and version from most common values
+    const status = ConfigurationExtractor.inferString(
+      resources,
+      'status'
+    ) as ImplementationGuideStatus;
+    const version = ConfigurationExtractor.inferString(resources, 'version');
+    if (status) {
+      config.config.status = status;
+    }
+    if (version) {
+      config.config.version = version;
+    }
+    return config;
+  }
+
+  static inferCanonical(resources: Package): string {
+    // url may be present as a CaretValueRule, path = '', caret path = 'url'
+    const potentialCanonicals: string[] = [];
+    [...resources.profiles, ...resources.extensions].forEach(resource => {
+      const urlRule = resource.rules.find(rule => {
+        return (
+          rule instanceof ExportableCaretValueRule && rule.path === '' && rule.caretPath === 'url'
+        );
+      }) as ExportableCaretValueRule;
+      if (urlRule) {
+        potentialCanonicals.push(
+          urlRule.value.toString().replace(/\/StructureDefinition\/[^/]+$/, '')
+        );
+      }
+    });
+    resources.valueSets.forEach(valueSet => {
+      const urlRule = valueSet.rules.find(rule => {
+        return (
+          rule instanceof ExportableCaretValueRule && rule.path === '' && rule.caretPath === 'url'
+        );
+      }) as ExportableCaretValueRule;
+      if (urlRule) {
+        potentialCanonicals.push(urlRule.value.toString().replace(/\/ValueSet\/[^/]+$/, ''));
+      }
+    });
+    resources.codeSystems.forEach(codeSystem => {
+      const urlRule = codeSystem.rules.find(rule => {
+        return (
+          rule instanceof ExportableCaretValueRule && rule.path === '' && rule.caretPath === 'url'
+        );
+      }) as ExportableCaretValueRule;
+      if (urlRule) {
+        potentialCanonicals.push(urlRule.value.toString().replace(/\/CodeSystem\/[^/]+$/, ''));
+      }
+    });
```

Comment:
The three blocks of code above are all exactly the same except for the variable they iterate over and the regular expression that they use.  I think you could just combine them all:
```ts
    [...resources.profiles, ...resources.extensions, ...resources.valueSets, ...resources.codeSystems].forEach(resource => {
      const urlRule = resource.rules.find(rule => {
        return (
          rule instanceof ExportableCaretValueRule && rule.path === '' && rule.caretPath === 'url'
        );
      }) as ExportableCaretValueRule;
      if (urlRule) {
        potentialCanonicals.push(
          urlRule.value.toString().replace(/\/(StructureDefinition|ValueSet|CodeSystem)\/[^/]+$/, '')
        );
      }
    });
```